robot/demos/SkidSteerServos/SkidSteeringDC.ino

```diff
@@ -0,0 +1,93 @@
+#include <ArduinoBlue.h>
+#include <SoftwareSerial.h>
+#include <Servo.h>
+
+int prevThrottle = 49;
+int prevSteering = 49;
+int throttle, steering;
+// Initialize Left & Right Velocity variables to Rest
+int VR = 127;
+int VL = 127;
+SoftwareSerial bluetooth(7, 8);
+
+ArduinoBlue phone(bluetooth);
+
+void setup() {
+  // Attach Servos to pins:
+  // Right side servos
+  DC_PIN_1 = 9;
+  DC_PIN_2 = 10;
+  DC_PIN_3 = 11;
+  
+  //Left side Servos
+  DC_PIN_4 = 3;
+  DC_PIN_5 = 5;
+  DC_PIN_6 = 6;
+
+  // initialize serial communications at 9600 bps:
+  Serial.begin(9600); 
+  bluetooth.begin(9600);
+
+  //activate pull-up resistor on the push-button pin
+  pinMode(buttonPin, INPUT_PULLUP); 
+  
+  Serial.println("setup complete");
+}
+
+float deg = 0;
+
+
+// New map function that supports float variables.
+ float mapFloat(float x, float in_min, float in_max, float out_min, float out_max) {
```

Comment:
Similar to what I said about the variable definition, it's nicer to either define your functions right before setup()... Or, what I and lots of others like to do, is to simply declare the function right before setup() and then actually define it after loop(). This hides away the details until the person trying to read your code feels the need to delve deeper and understand what's really going on with the function.

The forward declaration is standard C practice and it seems like it's not always necessary in Arduino to declare it before if you define it later, but I like to do so anyway.

Reply:
I think when we define our own standards for C/C++/Arduino we should use forward declarations and define the functions at the bottom always.